vllm/model_executor/models/granitemoe.py

```diff
@@ -92,13 +92,20 @@ def __init__(self,
                                 tp_size=tp_size,
                                 prefix=f"{prefix}.experts")
 
+        self.tp_size = tp_size if tp_size is not None else 1
+
     def forward(self, hidden_states: torch.Tensor) -> torch.Tensor:
         # NOTE: hidden_states can have either 1D or 2D shape.
         orig_shape = hidden_states.shape
         hidden_states = hidden_states.view(-1, self.hidden_size)
         # router_logits: (num_tokens, n_experts)
         router_logits, _ = self.gate(hidden_states)
         final_hidden_states = self.experts(hidden_states, router_logits)
+
+        if self.tp_size > 1:
+            final_hidden_states = self.experts.maybe_all_reduce_tensor_model_parallel(  # noqa E501
+                final_hidden_states)
+
```

Comment:
why did you need to change granitemoe?

Reply:
This is the same change that @varun-sundar-rabindranath  applied to the other MoE models.  I figured we needed it for granitemoe.py too.  Although I just tested with `PowerMoe-3b` and I seem to get reasonable results without it.

Reply:
I'll take this out for now and we can add back in later if it turns out we need it.